### DISCUSSION TITLE
Need help with RunReactants() - Never get any products

### DISCUSSION THREAD

**Discussion by tylerautera** [rdkit/rdkit#5766] [Q&A]
I have a list of mapped reactions (in SMILES) that I am attempting to utilize with RunReactants. However, the method never returns any results. In some cases (during troubleshooting), the reactant(s) passed into the RunReactants method are _the same_  as the reaction template itself, yet still no results. 

@greglandrum I need your help!

```
targ_smiles = "C(CCC(=O)O)CC(=O)O"
targ_mol = Chem.MolFromSmiles(targ_smiles)

template = '[O:1]=[C:2]([O-:3])[CH2:4][CH2:5][CH2:6][CH2:7][C:8](=[O:9])[O-:10]>>[OH2:3].[O:1]=[CH:2][CH2:4][CH2:5][CH2:6][CH2:7][C:8](=[O:9])[O-:10]'
reaction = AllChem.ReactionFromSmarts(template) 
products = reaction.RunReactants([targ_mol])
# Returns no products

template = '[O:1]=[CH:2][CH2:3][CH2:4][CH2:5][CH2:6][C:7](=[O:8])[O-:9]>>[OH:1][CH2:2][CH2:3][CH2:4][CH2:5][CH2:6][C:7](=[O:8])[O-:9]'
reaction = AllChem.ReactionFromSmarts(template) 
products = reaction.RunReactants([targ_mol])
# Returns no products

template = '[CH3:1][CH2:2][CH2:3][CH2:4][CH2:5][C:6](=[O:7])[O-:8].[CH3:9][NH3+:10]>>[CH3:1][CH2:2][CH2:3][CH2:4][CH2:5][C:6](=[O:7])[NH:10][CH3:9].[OH2:8]'
reaction = AllChem.ReactionFromSmarts(template) 
products = reaction.RunReactants([targ_mol, Chem.MolFromSMiles('[CH3:9][NH3+:10]')])
# Returns no products

template = ''[CH3:1][CH2:2][CH2:3][CH2:4][CH2:5][CH2:6][CH2:7][CH2:8][CH2:9][CH2:10][CH2:11][CH2:12][CH2:13][CH2:14][CH2:15][C:16](=[O:17])[O-:18]>>[CH3:1][CH2:2][CH2:3][CH2:4][CH2:5][CH2:6][CH2:7][CH2:8][CH2:9][CH2:10][CH2:11][CH2:12][CH2:13][CH2:14][CH2:15][CH:16]=[O:17].[OH2:18]''
reaction = AllChem.ReactionFromSmarts(template) 
products = reaction.RunReactants([targ_mol])
# Returns no products

**Comment by manajit-das** [comment]
Hi, @tylerautera  
I tried the first reaction you have mentioned. The error is in the mismatch between the input molecule (it is a neutral di carboxylic acid) and the reaction template (here the input has negatively charged species). Hence RunReactants doesn't result any output. When I changed the reactant to anionic form it works. Here is the code
![reaction](https://user-images.githubusercontent.com/59758360/202090431-6eff8d1b-f3a6-4a99-adb6-7fe9998ed434.png)

You can further read here: https://github.com/rdkit/rdkit-tutorials/blob/master/notebooks/003_SMARTS_ReactionsExamples.ipynb
Hope it helps.

**Reply by tylerautera** [reply]
@manajit-das thank you for the reply. I think this makes sense. I was doing some research yesterday and realized that I may need to change my approach here -- namely, focusing in on determining the reaction centers, then creating a smaller, more generalized template around the reactive atoms.
